### PR TITLE
Challenge 15

### PR DESCRIPTION
**Pull request by maddieamie**
[ x] Top-level README “Table of Contents” is updated
[ x] README for this challenge is complete
- [x ] Summary, Description, Approach & Efficiency, Solution
- [ x] Link to code
[ x] Feature tasks for this challenge are completed
[ x] Unit tests written and passing
- [x ] “Happy Path” - Expected outcome
- [ x] Expected failure
- [ x] Edge Case (if applicable/obvious)

### CODE REVIEW

File: python/binary_search_tree/binary_search_tree.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            raise IndexError("Stack is empty")
+
+    def size(self):
+        return len(self.items)
+
+    def __str__(self):
+        return str([node.data for node in self.items[::-1]])
+
+
+class BinaryTree(Queue, Stack):
```
Comment (brookr) [inline python/binary_search_tree/binary_search_tree.py:67]: Inheritance, or composition?

File: python/binary_search_tree/binary_search_tree.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+        self.visited = False
```
Comment (brookr) [inline python/binary_search_tree/binary_search_tree.py:6]: what is this for?

File: python/binary_search_tree/binary_search_tree.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+class BinaryTree(Queue, Stack):
+
+    def __init__(self):
+        super().__init__()
+        self.search_queue = Queue()
+        self.list_stack = Stack()
+        self.root = None
+
+    def pre_order(self) -> list:
```
Comment (brookr) [inline python/binary_search_tree/binary_search_tree.py:75]: consider recursion for these traversals, it's lovely.
